### PR TITLE
fix: spacing bugs

### PR DESCRIPTION
**Pull request by jnm2377**
Fixes spacing issues in image clusters (resulting from improper use of rows), and overrides Gatsby vertical spacing issues in colors page and SwatchPalette styles.

### CODE REVIEW

**Comment by jnm2377** [comment]
cc: @mjabbink @sadek @rcwakelin Not sure who should be reviewing this.

specifically fixes:
- bugs seen in the ColorGrid & SwatchPalette in colors page: https://github.ibm.com/brand/design-language/issues/67
- image cluster spacing in IBM Logos/8-bar page: https://github.ibm.com/brand/design-language/issues/68#issuecomment-15854450
- image cluster spacing in elements/2x-grid page: https://github.ibm.com/brand/design-language/issues/68#issuecomment-15854837
- image cluster spacing in photography/overview page
- image cluster spacing in colors page

File: src/components/SwatchPaletteWidget/_swatch-palette-widget.scss [hunk 1]
```diff
@@ -90,3 +90,7 @@
     height: 100%;
   }
 }
+
+[class*='--col'].#{$prefix}--swatch-palettes > * + * {
```
Comment (vpicone) [inline src/components/SwatchPaletteWidget/_swatch-palette-widget.scss]: If a 'stack' doesn't have different types of components, there's no need for the loose selector, we can just define `--space` with the child.

```suggestion
.#{$prefix}--swatch-palette {
```

**Comment by vpicone** [comment]
Does this one have issues to go along with it?

**Comment by jnm2377** [comment]
@vpicone the issues that were opened were all in the enterprise repo.

**Comment by jnm2377** [comment]
Rel: #349

**Comment by jnm2377** [comment]
@rc-wakelin @mjabbink what designer should be reviewing this PR?

**Comment by rcwakelin** [comment]
@sadek can you take review this ticket?

**Comment by jnm2377** [comment]
@sadekbazaraa could you review this?

**Comment by sadekbazaraa** [comment]
@jnm2377 I still see various spacing above captions as I scroll down multiple pages
https://deploy-preview-350--design-language-website.netlify.com/design/language/elements/2x-grid
https://deploy-preview-350--design-language-website.netlify.com/design/language/elements/ibm-logos/8-bar

– The spacing on the color palette page seems fixed
– Photography clusters seem good in general

I noticed this cluster on the 8-bar page that is off

Screen Shot 2019-11-22 at 4 23 53 PM

**Comment by mjabbink** [comment]
Spacing to tight @sadek and @jeanservaas can confirm spacing but it needs to be more space.

Screen Shot 2019-11-27 at 9 57 42 PM

**Comment by sadekbazaraa** [comment]
@jnm2377 @mjabbink the original design had 64px of space after the video. Under the new spacing scheme, I believe the spacing has been bumped up to 80px which would be similar to the 
spacing after other videos such as:
https://deploy-preview-350--design-language-website.netlify.com/design/language/elements/2x-grid

Screen Shot 2019-12-02 at 11 00 07 AM

This page also has a bit of a unique pattern relative to all the other pages so I'd be fine with a custom override or additional spacer in this case.

**Comment by jnm2377** [comment]
@sadekbazaraa @mjabbink I some of the spacing issues you pointed out. I _think_ I fixed everything, but I really could only fix obvious image/caption issues. Some other issues may not appear wrong to me because I don't know the specs of all these pages. Without specific issues with expected specs for a specific page, I have no way of knowing what is a visual bug.

For review: the pages I updated were:

- app-icons/design
- icons/design
- philosophy/pov
- ibm-logos/* (all of them)

**Comment by mjabbink** [comment]
Is there a preview link to look over?

**Comment by jnm2377** [comment]
@mjabbink https://deploy-preview-350--design-language-website.netlify.com/

Btw you'll  always find the latest preview links at the bottom of the page under the `checks` accordion item. For IDL it's the Netlify deploy preview, but I believe Carbon and Gatsby have a different tool they use for deploy previews (still under checks though).

Screen Shot 2019-12-04 at 9 43 32 AM

**Comment by mjabbink** [comment]
In Philosophy page the space between the paragraph is now too large. It should just be a full return in the text field. Just like here:https://www.ibm.com/design/language/philosophy/pov


Screen Shot 2019-12-04 at 10 16 37 AM

**Comment by mjabbink** [comment]
the spacing above caption should be 16px. @jeanservaas can you help with this. 

current in preview link is too large:
Screen Shot 2019-12-04 at 10 20 11 AM

**Comment by mjabbink** [comment]
@alisonjoseph @jnm2377 there are still major spacing issues. I recommend meeting and going over these in person. Capturing them in an issue would take too long.

**Comment by jnm2377** [comment]
@mjabbink I fixed the homepage, image and philosophy spacing issues. Still have to work on updating the mini image galleries to actually use the image gallery component. It might be a bit trickier than I anticipated or require a different design since we don't have any offset columns as part of the image gallery component. 

referring to this page:
Screen Shot 2019-12-05 at 4 13 21 PM

If we want this design specifically, it might require just overriding the current row spacing instead of using the image gallery component, or enhancing the image gallery component in gatsby, which might take a while.

I think it's worth merging this PR so the IDL site will at least have those major visual bugs fixed, since this is a minor bug. This PR contains the fix for all the colors pages and image clusters that didn't have margin, which has been on the current IDL site for weeks now..

**Comment by jnm2377** [comment]
cc: @sadekbazaraa see above comment as well.

**Comment by sadekbazaraa** [comment]
@jnm2377 I agree that we should merge and then revisit the larger issues around having flexibility with the image gallery component. Let's not hold up because of the spacing here.

**Comment by jnm2377** [comment]
@mjabbink have you had a chance to look at this again? I added a temporary override to the photography page gallery rows so that they have 32px margin top. This should be good to go now. I think other issues can be captured in a different PR.

**Comment by jnm2377** [comment]
@mjabbink is this good to go? I'd like to get this PR merged before I'm OOO for the holidays (in 2 days).

**Comment by mjabbink** [comment]
👍

**Comment by mjabbink** [comment]
Let’s get this deployed!

**Pull request merged**